### DISCUSSION TITLE
History feature & Duplicate goto page navigations

### DISCUSSION THREAD

**Discussion by jack-hyde-wk** [paoloricciuti/sveltekit-search-params#35] [Q&A] [answered]
I am struggling to understand this section of code. I was tracking down why my page was being navigated to twice whenever a param changed. 
https://github.com/paoloricciuti/sveltekit-search-params/blob/cff3b8c4d36ce06f3ed25f46414ca63ecf5ae76d/src/lib/sveltekit-search-params.ts#L195-L203
If I disable history, this fixes the issue. If I change it to an if else, this fixes the issue as well. Does the history feature require visiting the page twice like this?

Note this is particular code block is duplicated twice for queryParameters and queryParams. 

And thanks for the project, appreciate how simple it made dealing with everything as a svelte newbie.

**Comment by paoloricciuti** [comment, marked as answer]
It's actually not visiting the page twice...the first time is setting the query param without adding an history entry, the second one is just pushing again to the current url to add an history entry
